pkg/kubectl/cmd/resource/get.go

```diff
@@ -542,6 +542,11 @@ func (options *GetOptions) watch(f cmdutil.Factory, cmd *cobra.Command, args []s
 				return false, nil
 			}
 
+			if e.Type == watch.Deleted {
+				e.Object.(*unstructured.Unstructured).
+					Object["status"].(map[string]interface{})["phase"] = "Deleted"
```

Comment:
This isn't how we want to do this… this modifies data in the incoming object and only works for pods. Rather than modifying data in the object, we could consider displaying it differently if the event type was a delete event, but that would have to be part of the info passed to the printer

Reply:
Thanks. I was doubtful about this and had a thought of sending an extra argument to the printer but since that would require changes to the signature of a lot of functions, I thought I could get away with modifying the incoming data.
Is there any other way to pass extra info to the printer without introducing an extra argument to `PrintObj()`?

I thought of adding `type PrintExtraInfo map[string]interface{}` in printers and pass the extra object to all the `PrintObj()`. This `PrintExtraInfo` would have `event` type data for this particular case. But since this would require a lot of signature modification, I need a confirmation before I do it.